### ISSUE TITLE
Blank Window on Start

### ISSUE THREAD

**Issue by iwoloschin** [twolfson/google-music-electron#29]
Using v2.5.0 I'm getting a blank window on start.  Nothing seems to be wrong with the window itself, menus work and I can move/resize/minimize/etc the window, it's almost like it just isn't loading the Google Music page.  Happy to provide log files or anything else that might be useful, just not sure where any of that would be.

I'm running Funtoo Linux with Node v5.0.0 (this did just update from something else...curious if this could be a factor?).  Not sure of any other dependencies that would be useful to know.

**Comment by twolfson** [comment]
To clarify, is it a blank window when you initially launch or is it after the login screen?

**Comment by iwoloschin** [comment]
When I initially launch, there is no login screen.  It's kind of like
opening a new tab in Chrome, except there's no address bar to type anything
into.

**Comment by twolfson** [comment]
If you go to the Developer Tools console, are there any errors? (ctrl+shift+i or View -> Developer -> Toggle Developer Tools)

**Comment by iwoloschin** [comment]
The only thing that looks like an error shows:

/deep/ combinator is deprecated. See
https://www.chromestatus.com/features/6750456638341120 for more details.

Separately, it looks like the "Elements" tab is showing a very blank page:

<html>
  <head></head>
  <body></body>
</html>

Lots of other things to look at there but I'm not familiar with the
Developer Console, so not sure if there's anything else to check?

**Comment by twolfson** [comment]
What are the versions of everything in the "Help -> About google-music-electron"?

**Comment by iwoloschin** [comment]
google-music-electron

Version: 2.5.0
Electron version: 0.34.5
Node.js version: 4.1.1
Chromium version: 45.0.2454.85

**Comment by twolfson** [comment]
Hmm, everything seems fine. Can you paste the logs from `google-music-electron` into a gist (gist.github.com)?

**Comment by iwoloschin** [comment]
Sorry, busy week, finally have a chance to look at this again.  If I run google-music-electron from the CLI I only get two lines:

2015-12-20T01:19:01.499Z - info: Logger initialized. Writing info/warnings/errors to stdout. Writing all logs to "/home/iwoloschin/.config/google-music-electron/verbose.log"
2015-12-20T01:19:01.579Z - info: App ready. Opening Google Music window height=920, icon=/usr/lib64/node_modules/google-music-electron/lib/assets/icon-32.png, preload=/usr/lib64/node_modules/google-music-electron/lib/browser.js, skip-taskbar=undefined, use-content-size=true, width=1024, http_parser=2.5.0, node=4.1.1, v8=4.5.103.29, uv=1.7.4, zlib=1.2.8, ares=1.10.1-DEV, modules=46, openssl=1.0.2d, electron=0.34.5, atom-shell=0.34.5, chrome=45.0.2454.85, version=2.5.0

**Comment by twolfson** [comment]
Hmm, very strange indeed. I'm going to create a standalone gist for you to download to see if we can get a basic electron app running.

**Comment by twolfson** [comment]
Alright, I have created a gist. Can you follow the steps listed here and let know the results (you should see Google load):

https://gist.github.com/twolfson/5df0d2a9653be92dbec0#file-readme-md

**Comment by iwoloschin** [comment]
Same, just a blank, empty Electron window.

**Comment by twolfson** [comment]
Hmm, not sure what's wrong but it sounds like an issue between Electron and your OS. I'm going to search Electron's issues and maybe open a new one.

**Comment by twolfson** [comment]
Soo, it looks like you had `google-music-electron` running properly in #14. Was it working before the `node@5.0.0` upgrade?

**Comment by iwoloschin** [comment]
It was definitely working before node@5.0.0, but I haven't had much "free computer time" at home, so I haven't been using google-music-electron for a bit at home.  During that time I know that at least node was changed, not sure about anything else.

FWIW, it's working great on my OS X laptop at work...just not at home on Funtoo Linux.

**Comment by twolfson** [comment]
Are you using a node version management tool like `nvm` or `n`? If so, I would be interested in seeing what happens when you switch back to something like `0.10` or `4.2`

https://github.com/creationix/nvm

https://github.com/tj/n

> I personally use `n` since it doesn't do anything magical -- it overwrites existing `node`/`npm` installations rather than fancy `PATH` alterations.

**Comment by iwoloschin** [comment]
No, to be honest I don't know much about node in general, very much a dumb user in that area.  General Linux knowledge, sure, but until now I've been able to happily copy/paste and make node things work without thinking :).

I can easily (I think?) downgrade node via Portage, though it might be a bit odd if there's any other things on my system that have node dependencies (Atom is the only other thing I can think of though).

**Comment by twolfson** [comment]
Yea, I don't think this path will lead anywhere -- I was more curious than anything.

Is Atom working without any issues?

**Comment by twolfson** [comment]
Also, do you have any proxies set up on your network that need to be used and might be causing networking problems?

**Comment by iwoloschin** [comment]
Atom is working, though again, not really sure the dependencies there as I used my package manager (Portage) and a 3rd party overlay to set that up.

No proxies, pretty basic home internet setup.  Google Music works fine in Chrome, just seems that Electron windows are unhappy.

**Comment by twolfson** [comment]
Yea, that's really strange. Alright, I am out of ideas so let's open an issue on Atom =/

**Comment by twolfson** [comment]
Electron, I meant Electron =P

**Comment by twolfson** [comment]
Alright, I have opened up https://github.com/atom/electron/issues/3878. Thanks for the bug report =)

**Comment by iwoloschin** [comment]
Awesome.  I bet it'll wind up being something stupid on my end, but at least it isn't something glaringly obviously stupid?

I'll keep an eye on the Electron issue as well.

**Comment by aneutrals** [comment]
Fwiw, I'm also having the same issue. Totally blank screen. Distro is Arch.

Output of "~/.config/google-music-electron/verbose.log":

```
{"level":"info","message":"Logger initialized. Writing info/warnings/errors to stdout. Writing all logs to \"/home/user/.config/google-music-electron/verbose.log\"","timestamp":"2016-01-02T04:02:44.346Z"}
{"argv":["/usr/lib/node_modules/google-music-electron/node_modules/electron-prebuilt/dist/electron","/usr/lib/node_modules/google-music-electron"],"level":"debug","message":"CLI arguments received","timestamp":"2016-01-02T04:02:44.350Z"}
{"level":"debug","message":"CLI options overriding config","timestamp":"2016-01-02T04:02:44.384Z"}
{"playpause-shortcut":"mediaplaypause","next-shortcut":"medianexttrack","previous-shortcut":"mediaprevioustrack","level":"debug","message":"Generated starting config options","timestamp":"2016-01-02T04:02:44.394Z"}
{"options":{"height":920,"icon":"/usr/lib/node_modules/google-music-electron/lib/assets/icon-32.png","preload":"/usr/lib/node_modules/google-music-electron/lib/browser.js","use-content-size":true,"width":1024},"processVersions":{"http_parser":"2.5.0","node":"4.1.1","v8":"4.5.103.29","uv":"1.7.4","zlib":"1.2.8","ares":"1.10.1-DEV","modules":"46","openssl":"1.0.2d","electron":"0.34.5","atom-shell":"0.34.5","chrome":"45.0.2454.85"},"version":"2.6.0","level":"info","message":"App ready. Opening Google Music window","timestamp":"2016-01-02T04:02:44.609Z"}
```

**Comment by twolfson** [comment]
@aneutrals Could you follow the debugging steps and see if you yield the same results (e.g. running the gist)?

**Comment by aneutrals** [comment]
@twolfson Still a blank screen when running the gist.

**Comment by twolfson** [comment]
Alright, going to comment on the issue. I suggest you watch it as well for updates =/

**Comment by twolfson** [comment]
@iwoloschin @aneutrals On https://github.com/atom/electron/issues/3878 they have requested that we try out the latest version of `electron`. Can you try cloning the gist again and running it?

https://gist.github.com/twolfson/5df0d2a9653be92dbec0#file-readme-md

**Comment by iwoloschin** [comment]
Electron opens and loads www.google.

**Comment by twolfson** [comment]
Ah, nice. I will make an upgrade release in `google-music-electron` by the end of the weekend =)

**Comment by twolfson** [comment]
Alright, we have upgraded `electron` and released it in `2.7.0`. To upgrade, run the following:

``` bash
npm install -g google-music-electron@latest
```

I'm going to pre-emptively close this issue. Let us know if there any issues.
